heron/ckptmgr/src/java/com/twitter/heron/ckptmgr/CheckpointManagerConfig.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.ckptmgr;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.twitter.heron.common.config.ConfigReader;
+
+public class CheckpointManagerConfig {
+  public static final String STORAGE_CONFIG = "heron.statefulstorage.config";
```

Comment:
Using an enum of supported key values, that includes their types and default values isn't really heavyweight, it's good practice. It leads to easier to understand code with fewer bug and a better user experience. By following the approach in SystemConfig we can have the following:

- Type checking of field values. This has caused bugs in the past. Here's one from last week #1841.
- A type-safe way to represent default values that is compile-time verified.
- Ability to confirm at startup if someone has mis-typed a key name, causing some functionality to silently be enabled/disabled.
- A centralized place to manage keys and their defaults
- A way for the consumer to have some way to know the type of a given value, that's not coupled to the specific logic of the value producer.

Let me know if you'd like to discuss the pros/cons more offline.